tests/conftest.py

```diff
@@ -50,3 +54,20 @@ async def client(app: FastAPI) -> AsyncGenerator[AsyncClient]:
 def mock_butler() -> Iterator[MockButler]:
     """Mock Butler for testing."""
     yield from patch_butler()
+
+
+@pytest.fixture(autouse=True)
+def clear_hips_cache() -> Iterator[None]:
+    """Clear HiPS dependency caches between tests."""
+    from datalinker.dependencies.hips import (
+        dataset_hips_list_dependency,
+        hips_list_dependency,
+    )
+
+    hips_list_dependency._hips_list = None
+    dataset_hips_list_dependency._hips_lists = {}
```

Comment:
It feels to me like these should be methods. Maybe even a method on the base class so that all the dependencies provide the same API to clear the cache.